### PR TITLE
feat(BDropdown)!: replace "container" prop with "teleportTo"

### LINKED ISSUE #1799: useVModel => defineModel upgrade
### Clear and concise description of the problem

We want to upgrade to defineModel. useVModel syntax needs to adjust to this

```ts
const props = withDefaults(defineProps<{modelValue?: string}>(), {modelValue: '123'})
const emit = defineEmits<{ 'update:modelValue': [string]}>()

const modelVal = useVModel(props, 'modelValue', emit, {passive: true})
```
=> 
```ts
const modelVal = defineModel<string>({default: '123'})
```

There shouldn't be any unexpected breaking changes. I'm pretty sure the two should behave identically

Lastly, I believe this can adjust the workflow for BFormInput. We can explicitly define the modifiers, and create more straightforward syntax for modifiers. https://vuejs.org/api/sfc-script-setup.html#modifiers-and-transformers 

```ts
const [modelValue, modelModifiers] = defineModel<string, 'trim' | 'number'>({
  // get() omitted as it is not needed here
  set(value) {
    // if the .trim modifier is used, return trimmed value
    if (modelModifiers.trim) {
      return value.trim()
    }
    if(modelModifiers.number) {
      //
    }
    // otherwise, return the value as-is
    return value
  }
})
```

etc etc

### PR DESCRIPTION
**Pull request by VividLemon**
feat(BDropdown): add teleportDisabled prop

feat(BPopover)!: replace "container" prop with "teleportTo"

feat(BPopover): add teleportDisabled prop

feat: export component prop types

refactor: use defineModel > useVModel fixes #1799

fix(generics): use generic constraints for BTable & BTableLite

refactor: move all props into ComponentProps.ts -> export types & for future global options overhaul

# Describe the PR

A clear and concise description of what the pull request does.

## Small replication

A small replication or video walkthrough can help demonstrate the changes made. This is optional, but can help observe the intended changes. A mentioned issue that contains a replication also works.

## PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix :bug: - `fix(...)`
- [ ] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [ ] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [ ] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

### CODE REVIEW

File: packages/bootstrap-vue-next/src/components/BAccordion/BAccordion.vue [hunk 1]
```diff
@@ -8,33 +8,22 @@
...
+  id: undefined,
+})
 
 defineSlots<{
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   default?: (props: Record<string, never>) => any
 }>()
 
-const modelValue = useVModel(props, 'modelValue', emit, {passive: true})
+const modelValue = defineModel<string>({
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BAccordion/BAccordion.vue:24]:
```suggestion
const modelValue = defineModel<string | undefined>({
```

File: packages/bootstrap-vue-next/src/components/BButton/BButton.vue [hunk 2]
```diff
@@ -50,64 +49,46 @@ defineSlots<{
...
+})
 
 const emit = defineEmits<{
-  'click': [value: MouseEvent]
-  'update:pressed': [value: boolean]
+  click: [value: MouseEvent]
 }>()
 
-const pressedValue = useVModel(props, 'pressed', emit)
+const pressedValue = defineModel<boolean>('pressed', {default: undefined})
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BButton/BButton.vue:91]:
```suggestion
const pressedValue = defineModel<boolean | undefined>('pressed', {default: undefined})
```

File: packages/bootstrap-vue-next/src/components/BFormRadio/BFormRadio.vue [hunk 3]
```diff
@@ -23,68 +23,44 @@
...
+  value: true,
+})
 
 const slots = defineSlots<{
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   default?: (props: Record<string, never>) => any
 }>()
 
-const modelValue = useVModel(props, 'modelValue', emit, {passive: true})
+const modelValue = defineModel<RadioValue>({
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BFormRadio/BFormRadio.vue:61]:
```suggestion
const modelValue = defineModel<RadioValue | undefined>({
```

File: packages/bootstrap-vue-next/src/components/BFormRadio/BFormRadioGroup.vue [hunk 4]
```diff
@@ -93,7 +67,9 @@ defineSlots<{
   first?: (props: Record<string, never>) => any
 }>()
 
-const modelValue = useVModel(props, 'modelValue', emit, {passive: true})
+const modelValue = defineModel<RadioValue>({
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BFormRadio/BFormRadioGroup.vue:70]:
```suggestion
const modelValue = defineModel<RadioValue | null>({
```

**Comment by anrolmar** [comment]
If you don't have time to fix the PR, I could do it. Let me know if you need my help, @VividLemon

**Comment by VividLemon** [comment]
> If you don't have time to fix the PR, I could do it. Let me know if you need my help, @VividLemon 

Fixing the broken tests would be a big help. Tyvm

**Comment by VividLemon** [comment]
@anrolmar On second thought, I would prefer if you did not attempt to fix my branch. There are some things I want to change about the implementation

**Pull request closed without merging**